### PR TITLE
Added support for CURLOPT_SSL_VERIFYHOST

### PR DESCRIPTION
**Pull request by adampiotrowski**
This PR adds support for CURLOPT_SSL_VERIFYHOST option. Default value is
set to 2 according to documentation
(http://php.net//manual/pl/function.curl-setopt.php). There should not
be any BC issues.

### CODE REVIEW

**Comment by ewgRa** [comment]
+1

File: lib/Buzz/Client/AbstractStream.php [hunk 1]
```diff
@@ -19,7 +19,7 @@ public function getStreamContextArray(RequestInterface $request)
             'http' => array(
                 // values from the request
                 'method'           => $request->getMethod(),
-                'header'           => implode("\r\n", $request->getHeaders()),
+                'header'           => implode(PHP_EOL, $request->getHeaders()),
```
Comment (kriswallsmith) [inline lib/Buzz/Client/AbstractStream.php]: Why this change? The HTTP spec specifies `\r\n`.

File: test/Buzz/Test/Client/AbstractStreamTest.php [hunk 2]
```diff
@@ -30,7 +30,7 @@ public function testConvertsARequestToAContextArray()
         $expected = array(
             'http' => array(
                 'method'           => 'POST',
-                'header'           => "Content-Type: application/x-www-form-urlencoded\r\nContent-Length: 15",
+                'header'           => "Content-Type: application/x-www-form-urlencoded" . PHP_EOL . "Content-Length: 15",
```
Comment (kriswallsmith) [inline test/Buzz/Test/Client/AbstractStreamTest.php]: Again, why this change? Is it related to the purpose of the PR?

**Comment by adampiotrowski** [comment]
@kriswallsmith Do you want me to remove those 2 changes?

**Comment by kriswallsmith** [comment]
@adampiotrowski unless there's a justification i'm not aware of, yes please

**Comment by sverraest** [comment]
+1 to merge.
Currently not having this option is causing issue with a Load Balancer for testing purposes.
Any reason this has not been merged yet?

**Comment by ewgRa** [comment]
This is not merged, because @kriswallsmith is right. According to specs headers can be LF, but it is evil and something like support BC.

 The line terminator for message-header fields is the sequence CRLF.

**Comment by ewgRa** [comment]
@adampiotrowski ping, will you revert this changes?

**Comment by kriswallsmith** [comment]
see #195
